src/components/UserSelector.tsx

```diff
@@ -1,19 +1,51 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { User } from '../types/User';
+import { getUsers } from '../api/ApiMethods';
+
+type Props = {
+  setSelectedUser: (user: User) => void,
+  selectedUser: User | null,
+  setOpenedPost: (value: null) => void,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  setSelectedUser,
+  selectedUser,
+  setOpenedPost,
+}) => {
+  const [allUsers, setAllUsers] = useState<null | User[]>(null);
```

Comment:
```suggestion
  const [allUsers, setAllUsers] = useState<User[]>([]);
```
we can just use it as empty array instead of the null